tests/test_pixelScaleBoundedField.py

```diff
@@ -0,0 +1,114 @@
+# This file is part of afw.
+
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+
+import numpy as np
+
+import lsst.utils.tests
+import lsst.geom
+import lsst.afw.geom
+from lsst.afw.math import PixelScaleBoundedField
+
+
+class PixelScaleBoundedFieldTestCase(lsst.utils.tests.TestCase):
+    def setUp(self):
+        # Trivial WCS:
+        crpix = lsst.geom.Point2D(100, 100)
+        crval = lsst.geom.SpherePoint(0, 45, lsst.geom.degrees)
+        cdMatrix = lsst.afw.geom.makeCdMatrix(1*lsst.geom.arcseconds, 0*lsst.geom.degrees)
+        self.skyWcs = lsst.afw.geom.makeSkyWcs(crpix=crpix, crval=crval, cdMatrix=cdMatrix)
+
+        self.bbox = lsst.geom.Box2I(lsst.geom.Point2I(0, 0), lsst.geom.Extent2I(200, 200))
+        self.points = [crpix, ]
+        # ensure that the points are all Point2D
+        self.points.extend(lsst.geom.Box2D(self.bbox).getCorners())
+
+        self.boundedField = PixelScaleBoundedField(self.bbox, self.skyWcs)
+
+    def testConstruct(self):
+        self.assertEqual(self.boundedField.getBBox(), self.bbox)
+        self.assertEqual(self.boundedField.getSkyWcs(), self.skyWcs)
+        self.assertEqual(self.boundedField.getDefaultInverseScale(),
+                         1.0 / self.skyWcs.getPixelScale().asDegrees()**2)
+
+    def _computeExpected(self, points):
+        """Return an array with the expected result of evaluate(point)."""
+        expect = np.zeros(len(points), dtype=float)
+        for i, point in enumerate(points):
+            scale = self.skyWcs.getPixelScale(lsst.geom.Point2D(point)).asDegrees()
+            expect[i] = scale*scale / (self.skyWcs.getPixelScale().asDegrees())**2
```

Comment:
It's not "exactly the same": it's python instead of C++, which is often the only thing you can do for a test of mathematical functions.

What upstream changes are you afraid of?

We may also need/want to implement a vectorized version of the array evaluation code, which would require this existing python test code to pass.

Reply:
I do think what's here is better than nothing, as I do expect us to want to change the C++ implementation in the future, and in the meantime at least we're running the code and verifying that it doesn't throw or produce garbage.